js/script.js

```diff
@@ -0,0 +1,76 @@
+class Stopwatch {
+    constructor(display) {
+        this.running = false;
+        this.display = display;
+        this.reset();
+        this.print(this.times);
+    }
+
+    reset() {
+        this.times = {
+            minutes: 0,
+            seconds: 0,
+            miliseconds: 0
+        };
+    }
+
+    print() {
+        this.display.innerText = this.format(this.times);
+    }
+
+    format(times) {
+        return `${pad0(times.minutes)}:${pad0(times.seconds)}:${pad0(Math.floor(times.miliseconds))}`;
+    }
+
+    start() {
+        if (!this.running) {
+            this.running = true;
+            this.watch = setInterval(() => this.step(), 10);
+        }
+    }
+
+    step() {
+        if (!this.running) return;
+        this.calculate();
+        this.print();
+    }
+
+    calculate() {
+        this.times.miliseconds += 1;
+        if (this.times.miliseconds >= 100) {
+            this.times.seconds += 1;
+            this.times.miliseconds = 0;
+        }
+        if (this.times.seconds >= 60) {
+            this.times.minutes += 1;
+            this.times.seconds = 0;
+        }
+    }
+
+    stop() {
+        this.running = false;
+        clearInterval(this.watch);
+    }
+
+}
+
+const stopwatch = new Stopwatch(
+    document.querySelector('.stopwatch')),
+    startButton = document.getElementById('start'),
+    stopButton = document.getElementById('stop'),
+    resetButton = document.getElementById('reset');
+
+startButton.addEventListener('click', () => stopwatch.start());
```

Comment:
możesz to zapisać również w następujący sposób:
`startButton.addEventListener('click', stopwatch.start)`

Reply:
Nie chce mi zadziałać ten sposób.

Reply:
moj blad, być może this nie jest dobrze przekazane :)